File: csharp/log-levels/LogLevels.cs
using System;
using System.Linq;

static class LogLine
{
    public static string Message(string logLine) =>
        logLine
            .Substring(logLine.IndexOf(":") + 1)
            .Trim();

    public static string LogLevel(string logLine)
    {
        var startIndex = logLine.IndexOf("[") + 1;
        var numChars = logLine.IndexOf("]") - startIndex;

        return logLine.Substring(startIndex, numChars).ToLower();
    }

    public static string Reformat(string logLine) =>
        $"{Message(logLine)} ({LogLevel(logLine)})";
}

File: cpp/log-levels/Iteration1/log_levels.cpp
#include <string>

namespace log_line {
    std::string message(std::string line) {
        return line.substr(line.find(": ") + 2);
    }

    std::string log_level(std::string line) {
        int start = line.find("[") + 1;
        int end = line.find("]") - 1;
        
        return line.substr(start, end);
    }

    std::string reformat(std::string line) {
        return message(line) + " (" + log_level(line) + ")";
    }
}
